### PR TITLE
Auth api

### PR DESCRIPTION
**Pull request by VedanthB**
- added auth 
- signup
- login
- logout 
- storing JTW token in LS and using it to access Private Routes

### CODE REVIEW

File: src/components/auth-pages/LoginCard.js [hunk 1]
```diff
@@ -1,84 +1,130 @@
...
+        setUserData({ ...userData, [name]: value })
+    }
 
-        <Link
-          to="/"
-          className="btn btn-link-cyan text-white mb-5 pl-0"
-          style={{ padding: 0 }}
+    return (
+        <form
+            style={{ margin: 'auto', backgroundColor: '#0000227c' }}
```
Comment (Ananyamadhu08) [inline src/components/auth-pages/LoginCard.js:25]: If you are repeatedly using a colour then you can use variables.

File: src/context/reducers/filterReducer.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { filtersConstants } from '../constants/filtersConstants'
+
+// sortBy: null,
```
Comment (Ananyamadhu08) [inline src/context/reducers/filterReducer.js:3]: can delete commented code.

File: src/utils/getProducts.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+export const getProducts = async (productsDispatch) => {
+  try {
+    console.log(productsDispatch);
+
+    const {
+      data: { products },
+      status,
+    } = await getProductsService();
+
+    console.log(status);
```
Comment (Ananyamadhu08) [inline src/utils/getProducts.js:16]: can delete the unwanted console logs.

**Review by Ananyamadhu08 — APPROVED** [review]
Good work Vedanth! I have suggested small changes please look into those, other than that looks good! Clean and understandable code you can go ahead and merge.

**Review by Mohitkumar33 — APPROVED** [review]

**Pull request merged**